design-proposals/edge-infra-day2-improvements.md

```diff
@@ -0,0 +1,357 @@
+  - `update_sources`: the field is deprecated, and should not be used anymore.
+  - The resource now supports only Create, Read and Delete operations. Update 
+    is not allowed anymore. The resource is now read-only from the 
+    Northbound APIs perspective.
+- **Instance**:
+  - `desired_os`: field is removed, we don't drive the Immutable OS day 2 
+    workflow from this field anymore.
+  - `current_os`: the field can now be changed from EIM northbound APIs 
+    (REST), only before the EN is provisioned. After that, the field is 
+    handled internally by EIM, and will be updated based on the current OS 
+    installed.
+  - `installed_packages`: this field is used to track packages that are 
+    actually installed in the Edge Node at runtime. For Immutable OSes, we 
+    expect this to be equal to the content of `installed_packages` in the 
+    OS Profile. For Mutable OSes, this could change during the lifetime of 
+    the Edge Node.
+  - `new_versions`: signals that new updates are available. This field doesn't 
+    tell anything about the fact that the update will be done in the next 
+    scheduled maintenance. The field is a string.
+    - mutable: list of packages ready to be upgraded on the Edge Node, 
+      empty if no new versions are available.
+    - immutable: latest EMT version, empty if the version is already latest.
+  - `next_update_policy`: The update policy that is applied during the 
+    maintenance schedule.
+  
+New Resources:
+
+- **OS Update Policy**: defines the Policy that must be applied during an 
+  update. This resource can be linked to multiple Instances, Regions and or 
+  Sites. Some fields and edges are valid for Mutable or Immutable OSes.
+  - `install_packages`: list of packages to be installed during the update. 
+    This field is used only for Mutable OSes.
+  - `update_sources`: 'DEB822 Source Format' for Debian style OSs, contains 
+    the sources of the packages to be installed during the update, if new 
+    sources needs to be added.
+  - `update_policy`: enum field, defines which policy to use:
+    - `UPDATE_POLICY_LATEST`: upgrade to latest version. 
+      - mutable: upgrade all packages to latest versions.
+      - immutable: upgrade the OS version to the latest.
+    - `UPDATE_POLICY_TARGET`: upgrade to a specific version.
+      - mutable: upgrade all packages to the specified version defined in 
+        the `install_package` list.
+      - immutable: upgrade the OS version to OS Version linked by the 
+        `target_os` edge.
+  - `target_os`: edge to the OS Profile that must be installed during the 
+    update. This field is used only for Immutable OSes.
+
+- **OS Update Job**: stores the information about the update job that run 
+  on the Edge Node. This resource is created when the update is started, 
+  and stores information about what happened in the Edge Node during the 
+  Update.
+  - `status`: stores the short description of what happened during the update.
```

Comment:
If nothing was updated, do we need to create an OS Update Job?

Reply:
To track that an update has happened (even if it was no-op), IMO we should create them.
To avoid too many instances, of these resources, we should define a retain policy for them.
I'll add ideas about retention of the OSUpdateJob.

Reply:
I'll add it to the implementation plan

Reply:
DOne